gammapy/image/tests/test_catalog.py

```diff
@@ -64,8 +64,31 @@ def test_catalog_image():
 def test_catalog_table():
     # Checks catalogs are loaded correctly
 
-    table_1fhl = catalog.catalog_table('1FHL')
+    table_1fhl = catalog_table('1FHL')
     assert len(table_1fhl) == 514
 
-    table_2fgl = catalog.catalog_table('2FGL')
+    table_2fgl = catalog_table('2FGL')
     assert len(table_2fgl) == 1873
+
+
+
+class TestCatalogImageEstimator(object):
+    def setup(self):
+        self.reference = SkyImage.empty(xref=18.0, yref=-0.6, nypix=41,
+                                        nxpix=41, binsz=0.1)
+        self.estimator = CatalogImageEstimator(reference=self.reference,
+                                               emin=1 * u.TeV,
+                                               emax=1E4 * u.TeV)
+    @requires_data('gammapy-extra')
+    def test_flux_gammacat(self):
+        from ...catalog import SourceCatalogGammaCat
+        catalog = SourceCatalogGammaCat()
+        result = self.estimator.run(catalog)
+
+        actual = result['flux'].data.sum()
+        selection = catalog.select_image_region(self.reference)
+
+        assert len(selection.table) == 3
+
+        desired = selection.table['spec_flux_above_1TeV'].sum()
+        assert_allclose(actual, desired, rtol=1E-3)
```

Comment:
Add a few more asserts that establish behaviour at the edge and for small sources (i.e. have pixel integration is done or not done), to make it easier to refactor this code in the future?
Really this is a special case of the model rendering that will also be done in likelihood fitting, and it would IMO be better if the API already deals with lists of models and is decoupled from `gammapy.catalog`.